Objects/regions.c

```diff
@@ -160,28 +195,76 @@ typedef struct _gc_runtime_state GCState;
 #define GC_PREV _PyGCHead_PREV
 #define FROM_GC(g) ((PyObject *)(((char *)(g))+sizeof(PyGC_Head)))
 
+static int
+is_immutable_region(Py_uintptr_t r)
```

Comment:
Why not use macros here?

Reply:
(for one thing you can incorporate the cast to `Py_uintptr_t` instead of having it at the call site)